### PR TITLE
Commands Transformation

### PR DESCRIPTION
**Pull request by Patryk-97**
Patryk Stępień 215886
Sebastian Pawełoszek 215844
Leonard Wojtczak 215923
Damian Rutkowski 215868

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/command/CommandCoordinatesVisitor.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import edu.kis.powp.jobs2d.command.line.Line2d;
+
+
+public class CommandCoordinatesVisitor implements Visitor {
+	
+	List<Line2d> lines = new ArrayList<>();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/CommandCoordinatesVisitor.java]: should be private

File: src/main/java/edu/kis/powp/jobs2d/command/CommandCoordinatesVisitor.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			driverCommand.accept(this);
+		}
+	}
+
+	public void visit(DriverCommand compoundCommand) {
+		if(compoundCommand instanceof ICompoundCommand) {
+			ICompoundCommand iCompoundCommand = (ICompoundCommand) compoundCommand;
+			visit(iCompoundCommand);
+		}
+	}
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/CommandCoordinatesVisitor.java]: There shouldn't be such a visit in visitor.

File: src/main/java/edu/kis/powp/jobs2d/command/transformations/TransformationManager.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+		for(int i = 0; i < lines.size(); i++) {
+			lines.get(i).setStartPosX(-lines.get(i).getStartPosX());
+			lines.get(i).setEndPosX(-lines.get(i).getEndPosX());
+		}
+
+		List<DriverCommand> commandList = buildCommandList(lines);
+
+		return new ImmutableComplexCommand(commandList);
+	}
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/transformations/TransformationManager.java]: All methods share some code (duplication). The code is not OCP. Adding new transformation requires modification in this class (containing other transformations).

**Review by kowallus — CHANGES_REQUESTED** [review]
Major code duplication in the transformation manager.

**Comment by kowallus** [comment]
please close, fix and reopen

**Comment by Patryk-97** [comment]
Dzień dobry, wydaje nam się, że rozwiązaliśmy problemy, które Pan wskazał.

**Pull request merged**